nixos/paperless: granian asginl dumped core while starting paperless

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

After the nixpkgs update, paperless starts, but granian doesn't start anymore and it occupies around 60% of the cpu.

I believe the problem is in some python compatibility and I don't know if it is granian specific or due to paperless.

paperless-web.service and paperless-consumer.sevice are both running and don't output any error message, but `journalctl -u paperless-web -e` output the long error message that I have reported below.

When it try to reach it with nginx, it gets the Bad Gateway message.

Thanks

p.s. It was working yesterday with the previous nixpkgs commit, I update it everyday so it should be something happened in the last commit of nixpkgs

### Steps to reproduce

This is my configuration for paperless:

```
      services.paperless = {
        user = "paperless";
        enable = true;
        address = ip;
        dataDir = dataDir;
        passwordFile = sopsSecret.path;
        settings = {
          PAPERLESS_URL = publicPath;
          PAPERLESS_SECRET_KEY = "placeholder";
          PAPERLESS_ALLOWED_HOSTS = "${service}.${domain},${hostName},${url},${ip},localhost";
          PAPERLESS_CONSUMER_IGNORE_PATTERN = [
            # ".DS_STORE/*"
            # "desktop.ini"
          ];
          PAPERLESS_AUTO_LOGIN_USERNAME = "User";
          PAPERLESS_CONSUMER_RECURSIVE = false;
          PAPERLESS_OCR_LANGUAGE = "ita+eng+dan";
          PAPERLESS_OCR_USER_ARGS = {
            optimize = 1;
            pdfa_image_compression = "lossless";
          };
        };
      };
```


### Expected behaviour

For paperless to be reacheable by nginx and working.

### Screenshots

_No response_

### Relevant log output

```console
Jun 05 13:31:33 romeo systemd[1]: paperless-web.service: Scheduled restart job, restart counter is at 37.
Jun 05 13:31:33 romeo systemd[1]: Started Paperless web server.
Jun 05 13:31:34 romeo paperless-web-start[147762]: [INFO] Starting granian (main PID: 147762)
Jun 05 13:31:34 romeo paperless-web-start[147762]: [INFO] Listening at: http://100.77.167.122:28981
Jun 05 13:31:34 romeo paperless-web-start[147762]: [INFO] Spawning worker-1 with PID: 147766
Jun 05 13:31:37 romeo systemd-coredump[147783]: Process 147766 (granian asginl ) of user 315 dumped core.
                                                
                                                Module libqpdf.so.29 without build-id.
                                                Module _core.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module _cffi_backend.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module _rust.abi3.so without build-id.
                                                Module md__mypyc.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module md.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module _psycopg.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module libkeyutils.so.1 without build-id.
                                                Module libkrb5support.so.0 without build-id.
                                                Module libcom_err.so.3 without build-id.
                                                Module libk5crypto.so.3 without build-id.
                                                Module libkrb5.so.3 without build-id.
                                                Module libgssapi_krb5.so.2 without build-id.
                                                Module pq.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module _speedups.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module libXdmcp.so.6 without build-id.
                                                Module libXau.so.6 without build-id.
                                                Module libsharpyuv.so.0 without build-id.
                                                Module libwebp.so.7 without build-id.
                                                Module libzstd.so.1 without build-id.
                                                Module libLerc.so.4 without build-id.
                                                Module libdeflate.so.0 without build-id.
                                                Module libxcb.so.1 without build-id.
                                                Module libimagequant.so.0.4 without build-id.
                                                Module libopenjp2.so.7 without build-id.
                                                Module libjpeg.so.62 without build-id.
                                                Module libtiff.so.6 without build-id.
                                                Module _imaging.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module libmagic.so without build-id.
                                                Module libffi.so.8 without build-id.
                                                Module hiredis.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module backend_c.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module _brotli.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module _cmsgpack.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module libyaml-0.so.2 without build-id.
                                                Module _yaml.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module libmpdec.so.4 without build-id.
                                                Module _setproctitle.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module liblzma.so.5 without build-id.
                                                Module libbz2.so.1 without build-id.
                                                Module libz.so.1 without build-id.
                                                Module libstdc++.so.6 without build-id.
                                                Module libjemalloc.so.2 without build-id.
                                                Module _granian.cpython-312-x86_64-linux-gnu.so without build-id.
                                                Module libgcc_s.so.1 without build-id.
                                                Stack trace of thread 147766:
                                                #0  0x00007fddae261ec8 _rjem_je_free_default (libjemalloc.so.2 + 0x1eec8)
                                                #1  0x00007fdd6c0ac3cb _ZNSt6vectorIN12_GLOBAL__N_110JSONParser10StackFrameESaIS2_EE12emplace_backIJNS1_14parser_state_eER4JSONEEERS2_DpOT_.isra.0 (libqpdf.so.29 + 0xac3cb)
                                                #2  0x00007fdd6c0b4fca _ZN12_GLOBAL__N_110JSONParser11handleTokenEv (libqpdf.so.29 + 0xb4fca)
                                                #3  0x00007fdd6c0b5540 _ZN12_GLOBAL__N_110JSONParser5parseEv.constprop.0 (libqpdf.so.29 + 0xb5540)
                                                #4  0x00007fdd6c0b5d03 _ZN4JSON5parseERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE (libqpdf.so.29 + 0xb5d03)
                                                #5  0x00007fdd6c0a58b9 _GLOBAL__sub_I_QPDFJob_json.cc (libqpdf.so.29 + 0xa58b9)
                                                #6  0x00007fddafdde63e call_init (ld-linux-x86-64.so.2 + 0x463e)
                                                #7  0x00007fddafdde73c _dl_init (ld-linux-x86-64.so.2 + 0x473c)
                                                #8  0x00007fddafddb586 _dl_catch_exception (ld-linux-x86-64.so.2 + 0x1586)
                                                #9  0x00007fddafde57ce dl_open_worker (ld-linux-x86-64.so.2 + 0xb7ce)
                                                #10 0x00007fddafddb4f1 _dl_catch_exception (ld-linux-x86-64.so.2 + 0x14f1)
                                                #11 0x00007fddafde5b5c _dl_open (ld-linux-x86-64.so.2 + 0xbb5c)
                                                #12 0x00007fddaf29379c dlopen_doit (libc.so.6 + 0x9379c)
                                                #13 0x00007fddafddb4f1 _dl_catch_exception (ld-linux-x86-64.so.2 + 0x14f1)
                                                #14 0x00007fddafddb623 _dl_catch_error (ld-linux-x86-64.so.2 + 0x1623)
                                                #15 0x00007fddaf293217 _dlerror_run (libc.so.6 + 0x93217)
                                                #16 0x00007fddaf293871 dlopen@GLIBC_2.2.5 (libc.so.6 + 0x93871)
                                                #17 0x00007fddaf8e4878 _PyImport_FindSharedFuncptr (libpython3.12.so.1.0 + 0x2e4878)
                                                #18 0x00007fddaf9e7abf _imp_create_dynamic (libpython3.12.so.1.0 + 0x3e7abf)
                                                #19 0x00007fddaf853284 cfunction_vectorcall_FASTCALL (libpython3.12.so.1.0 + 0x253284)
                                                #20 0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #21 0x00007fddaf88b26e object_vacall (libpython3.12.so.1.0 + 0x28b26e)
                                                #22 0x00007fddaf8e8c05 PyObject_CallMethodObjArgs (libpython3.12.so.1.0 + 0x2e8c05)
                                                #23 0x00007fddaf796a3b PyImport_ImportModuleLevelObject (libpython3.12.so.1.0 + 0x196a3b)
                                                #24 0x00007fddaf8440b7 cfunction_vectorcall_FASTCALL_KEYWORDS (libpython3.12.so.1.0 + 0x2440b7)
                                                #25 0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #26 0x00007fddaf88b26e object_vacall (libpython3.12.so.1.0 + 0x28b26e)
                                                #27 0x00007fddaf8e8c05 PyObject_CallMethodObjArgs (libpython3.12.so.1.0 + 0x2e8c05)
                                                #28 0x00007fddaf795fc5 PyImport_ImportModuleLevelObject (libpython3.12.so.1.0 + 0x195fc5)
                                                #29 0x00007fddafa05613 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x405613)
                                                #30 0x00007fddafa0ab6d PyEval_EvalCode (libpython3.12.so.1.0 + 0x40ab6d)
                                                #31 0x00007fddafa14a85 builtin_exec (libpython3.12.so.1.0 + 0x414a85)
                                                #32 0x00007fddaf8440b7 cfunction_vectorcall_FASTCALL_KEYWORDS (libpython3.12.so.1.0 + 0x2440b7)
                                                #33 0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #34 0x00007fddaf88b26e object_vacall (libpython3.12.so.1.0 + 0x28b26e)
                                                #35 0x00007fddaf8e8c05 PyObject_CallMethodObjArgs (libpython3.12.so.1.0 + 0x2e8c05)
                                                #36 0x00007fddaf796a3b PyImport_ImportModuleLevelObject (libpython3.12.so.1.0 + 0x196a3b)
                                                #37 0x00007fddafa05613 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x405613)
                                                #38 0x00007fddafa0ab6d PyEval_EvalCode (libpython3.12.so.1.0 + 0x40ab6d)
                                                #39 0x00007fddafa14a85 builtin_exec (libpython3.12.so.1.0 + 0x414a85)
                                                #40 0x00007fddaf8440b7 cfunction_vectorcall_FASTCALL_KEYWORDS (libpython3.12.so.1.0 + 0x2440b7)
                                                #41 0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #42 0x00007fddaf88b26e object_vacall (libpython3.12.so.1.0 + 0x28b26e)
                                                #43 0x00007fddaf8e8c05 PyObject_CallMethodObjArgs (libpython3.12.so.1.0 + 0x2e8c05)
                                                #44 0x00007fddaf796a3b PyImport_ImportModuleLevelObject (libpython3.12.so.1.0 + 0x196a3b)
                                                #45 0x00007fddafa05613 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x405613)
                                                #46 0x00007fddafa0ab6d PyEval_EvalCode (libpython3.12.so.1.0 + 0x40ab6d)
                                                #47 0x00007fddafa14a85 builtin_exec (libpython3.12.so.1.0 + 0x414a85)
                                                #48 0x00007fddaf8440b7 cfunction_vectorcall_FASTCALL_KEYWORDS (libpython3.12.so.1.0 + 0x2440b7)
                                                #49 0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #50 0x00007fddaf88b26e object_vacall (libpython3.12.so.1.0 + 0x28b26e)
                                                #51 0x00007fddaf8e8c05 PyObject_CallMethodObjArgs (libpython3.12.so.1.0 + 0x2e8c05)
                                                #52 0x00007fddaf796a3b PyImport_ImportModuleLevelObject (libpython3.12.so.1.0 + 0x196a3b)
                                                #53 0x00007fddafa05613 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x405613)
                                                #54 0x00007fddafa0ab6d PyEval_EvalCode (libpython3.12.so.1.0 + 0x40ab6d)
                                                #55 0x00007fddafa14a85 builtin_exec (libpython3.12.so.1.0 + 0x414a85)
                                                #56 0x00007fddaf8440b7 cfunction_vectorcall_FASTCALL_KEYWORDS (libpython3.12.so.1.0 + 0x2440b7)
                                                #57 0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #58 0x00007fddaf88b26e object_vacall (libpython3.12.so.1.0 + 0x28b26e)
                                                #59 0x00007fddaf8e8c05 PyObject_CallMethodObjArgs (libpython3.12.so.1.0 + 0x2e8c05)
                                                #60 0x00007fddaf796a3b PyImport_ImportModuleLevelObject (libpython3.12.so.1.0 + 0x196a3b)
                                                #61 0x00007fddaf8440b7 cfunction_vectorcall_FASTCALL_KEYWORDS (libpython3.12.so.1.0 + 0x2440b7)
                                                #62 0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #63 0x00007fddaf88b26e object_vacall (libpython3.12.so.1.0 + 0x28b26e)
                                                
                                                Stack trace of thread 147774:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147768:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147769:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147772:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147771:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147780:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147775:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147770:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147776:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147777:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147773:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147778:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                
                                                Stack trace of thread 147779:
                                                #0  0x00007fddaf29438e __futex_abstimed_wait_common (libc.so.6 + 0x9438e)
                                                #1  0x00007fddaf2a0268 __new_sem_wait_slow64.constprop.0 (libc.so.6 + 0xa0268)
                                                #2  0x00007fddaf747335 PyThread_acquire_lock_timed (libpython3.12.so.1.0 + 0x147335)
                                                #3  0x00007fddaf9a4ced acquire_timed (libpython3.12.so.1.0 + 0x3a4ced)
                                                #4  0x00007fddaf9a528c lock_PyThread_acquire_lock (libpython3.12.so.1.0 + 0x3a528c)
                                                #5  0x00007fddaf8b6645 method_vectorcall_VARARGS_KEYWORDS (libpython3.12.so.1.0 + 0x2b6645)
                                                #6  0x00007fddaf9ffaff _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3ffaff)
                                                #7  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #8  0x00007fddaf9fbc63 _PyEval_EvalFrameDefault (libpython3.12.so.1.0 + 0x3fbc63)
                                                #9  0x00007fddaf8b1ccb method_vectorcall (libpython3.12.so.1.0 + 0x2b1ccb)
                                                #10 0x00007fddaf863b5e thread_run (libpython3.12.so.1.0 + 0x263b5e)
                                                #11 0x00007fddaf717906 pythread_wrapper (libpython3.12.so.1.0 + 0x117906)
                                                #12 0x00007fddaf297e63 start_thread (libc.so.6 + 0x97e63)
                                                #13 0x00007fddaf31bb94 __clone (libc.so.6 + 0x11bb94)
                                                ELF object binary architecture: AMD x86-64
Jun 05 13:31:37 romeo paperless-web-start[147762]: [ERROR] Unexpected exit from worker-1
Jun 05 13:31:37 romeo paperless-web-start[147762]: [INFO] Shutting down granian
Jun 05 13:31:37 romeo systemd[1]: paperless-web.service: Main process exited, code=exited, status=1/FAILURE
Jun 05 13:31:37 romeo systemd[1]: paperless-web.service: Failed with result 'exit-code'.
Jun 05 13:31:37 romeo systemd[1]: paperless-web.service: Consumed 2.538s CPU time, 101.9M memory peak.
Jun 05 13:31:37 romeo systemd[1]: paperless-web.service: Scheduled restart job, restart counter is at 38.
Jun 05 13:31:37 romeo systemd[1]: Started Paperless web server.
Jun 05 13:31:38 romeo paperless-web-start[147792]: [INFO] Starting granian (main PID: 147792)
Jun 05 13:31:38 romeo paperless-web-start[147792]: [INFO] Listening at: http://100.77.167.122:28981
Jun 05 13:31:38 romeo paperless-web-start[147792]: [INFO] Spawning worker-1 with PID: 147797
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.31, NixOS, 25.11 (Xantusia), 25.11.20250603.c2a0396`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/1zw47fx5h4x65n914j4b9iz0j3v17aw0-source`


### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/granian/default.nix
+++ b/pkgs/development/python-modules/granian/default.nix
@@ -1,6 +1,5 @@
 {
   lib,
-  stdenv,
   fetchFromGitHub,
   rustPlatform,
   cacert,
@@ -14,7 +13,6 @@
   pytest-asyncio,
   websockets,
   httpx,
-  rust-jemalloc-sys,
   sniffio,
   nix-update-script,
 }:
@@ -31,6 +29,14 @@ buildPythonPackage rec {
     hash = "sha256-qJ65ILj7xLqOWmpn1UzNQHUnzFg714gntVSmYHpI65I=";
   };
 
+  # Granian forces a custom allocator for all the things it runs,
+  # which breaks some libraries in funny ways. Make it not do that,
+  # and allow the final application to make the allocator decision
+  # via LD_PRELOAD or similar.
+  patches = [
+    ./no-alloc.patch
+  ];
+
   cargoDeps = rustPlatform.fetchCargoVendor {
     inherit pname version src;
     hash = "sha256-swfqKp8AsxNAUc7dlce6J4dNQbNGWrCcUDc31AhuMmI=";
@@ -39,22 +45,6 @@ buildPythonPackage rec {
   nativeBuildInputs = with rustPlatform; [
     cargoSetupHook
     maturinBuildHook
-  ];
-
-  buildInputs = lib.optionals (stdenv.hostPlatform.isAarch64) [
-    # fix "Unsupported system page size" on aarch64-linux with 16k pages
-    # https://github.com/NixOS/nixpkgs/issues/410572
-    # only enabled on aarch64 due to https://github.com/NixOS/nixpkgs/pull/410611#issuecomment-2939564567
-    (rust-jemalloc-sys.overrideAttrs (
-      { configureFlags, ... }:
-      {
-        configureFlags = configureFlags ++ [
-          # otherwise import check fails with:
-          # ImportError: {{storeDir}}/lib/libjemalloc.so.2: cannot allocate memory in static TLS block
-          "--disable-initial-exec-tls"
-        ];
-      }
-    ))
   ];
 
   dependencies = [

--- a/pkgs/development/python-modules/granian/no-alloc.patch
+++ b/pkgs/development/python-modules/granian/no-alloc.patch
@@ -1,0 +1,50 @@
+diff --git a/Cargo.toml b/Cargo.toml
+index 4e6676f..1657d61 100644
+--- a/Cargo.toml
++++ b/Cargo.toml
+@@ -40,7 +40,6 @@ hyper = { version = "=1.6", features = ["http1", "http2", "server"] }
+ hyper-util = { version = "=0.1", features = ["server-auto", "tokio"] }
+ itertools = "0.14"
+ log = "0.4"
+-mimalloc = { version = "0.1.43", default-features = false, features = ["local_dynamic_tls"], optional = true }
+ mime_guess = "=2.0"
+ pem = "=3.0"
+ percent-encoding = "=2.3"
+@@ -56,15 +55,9 @@ tokio-stream = "0.1"
+ tokio-tungstenite = "=0.26"
+ tokio-util = { version = "0.7", features = ["codec", "rt"] }
+ 
+-[target.'cfg(not(any(target_env = "musl", target_os = "freebsd", target_os = "openbsd", target_os = "windows")))'.dependencies]
+-tikv-jemallocator = { version = "0.6.0", default-features = false, features = ["disable_initial_exec_tls"] }
+-
+ [build-dependencies]
+ pyo3-build-config = "=0.25"
+ 
+-[features]
+-mimalloc = ["dep:mimalloc"]
+-
+ [profile.release]
+ codegen-units = 1
+ debug = false
+diff --git a/src/lib.rs b/src/lib.rs
+index 9172842..6c41005 100644
+--- a/src/lib.rs
++++ b/src/lib.rs
+@@ -1,17 +1,3 @@
+-#[cfg(not(any(
+-    target_env = "musl",
+-    target_os = "freebsd",
+-    target_os = "openbsd",
+-    target_os = "windows",
+-    feature = "mimalloc"
+-)))]
+-#[global_allocator]
+-static GLOBAL: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
+-
+-#[cfg(feature = "mimalloc")]
+-#[global_allocator]
+-static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
+-
+ use pyo3::prelude::*;
+ use std::sync::OnceLock;
+ 

